src/axom/inlet/tests/inlet_writer.cpp

```diff
@@ -0,0 +1,253 @@
+// Copyright (c) 2017-2021, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "gtest/gtest.h"
+
+#include <cstdlib>
+
+#include <string>
+#include <vector>
+#include <memory>
+#include <fstream>
+#include <streambuf>
+
+#include "axom/inlet/Inlet.hpp"
+#include "axom/inlet/JSONSchemaWriter.hpp"
+#include "axom/inlet/tests/inlet_test_utils.hpp"
+
+using axom::inlet::Inlet;
+using axom::inlet::JSONSchemaWriter;
+using axom::sidre::DataStore;
+
+bool hasSchemaUtility()
+{
+  const static bool hasUtility =
+    std::system("/usr/bin/jsonschema --version") == 0;
```

Comment:
This seems brittle.
Is this typically available in unix? What about windows?

Is there a way we can add this via the build system, e.g. via axom's `devtools` and pass the path in through a `#define` in `axom/config.hpp`?

Perhaps as a follow-up PR?

Reply:
I have the path hardcoded here for security reasons (to avoid PATH injection attacks, though I'm not sure if this is a threat vector that Axom cares about).

I don't recall installing `jsonschema` (it just happened to be present), but I think it's available as a PyPI package that can be installed fairly easily.  I think it could also be made available through spack.